components/psa/services/psa_prot_internal_storage/COMPONENT_PSA_SRV_IMPL/pits_impl.cpp

```diff
@@ -0,0 +1,237 @@
+/* Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cstring>
+#include "KVMap.h"
+#include "KVStore.h"
+#include "TDBStore.h"
+#include "psa_prot_internal_storage.h"
+#include "pits_impl.h"
+#include "mbed_error.h"
+
+#ifdef   __cplusplus
+extern "C"
+{
+#endif
+
+using namespace mbed;
+
+#define PSA_ITS_FILENAME_LEN_BYTES      14          // Maximum length of filename we use for kvstore API.
+                                                    // uid: 6; delimiter: 1; pid: 6; str terminator: 1
+
+
+const uint8_t base64_coding_table[] = {
```

Comment:
the file names are `char*` why are we using `uint8_t` type during the whole flow and then casting?
i would suggest picking type the suite your needs in the first place.

Reply:
The cast is needed because of the const so we will have to cast anyway.

Reply:
this is not true - please double check